### PR TITLE
Dec API

### PR DESCRIPTION
**Pull request by PhilippSchenk**
Should be working great now

### CODE REVIEW

**Comment by FloH11110** [comment]
one thing you could do h-file:
leave the parameter list not empty, because this mean not defined, write void instead

File: source/DEC_API.c [hunk 1]
```diff
@@ -21,11 +21,15 @@
 #include "global.h"
 #include "DEC_API.h"
 
+static char dec_last = 0x00;
+
 void DEC_init(){
     LOG("rotencoder init\n");
+    AD1PCFGL       |= 0x1800; /*Set AN11, AN12 to Digital Pins*/
```
Comment (FloH11110) [inline source/DEC_API.c:28]: this should be in the global or ?
@SebastianWolfarth

Comment (SebastianWolfarth) [inline source/DEC_API.c:28]: It is done in main(), but its no problem if its done here again...

File: source/DEC_API.c [hunk 2]
```diff
@@ -21,11 +21,15 @@
 #include "global.h"
 #include "DEC_API.h"
 
+static char dec_last = 0x00;
+
 void DEC_init(){
     LOG("rotencoder init\n");
```
Comment (SebastianWolfarth) [inline source/DEC_API.c]: Please use a dedicated logging level e.g. "DEC_LOG()" or leave it out. I think its the only LOG() you used so you can leave it out.

File: source/DEC_API.c (hunk 1 above)
Comment (PhilippSchenk) [inline source/DEC_API.c:28]: @FlorianHeidecker parameter list updated thx for feedback
re AD1PCFGL better safe than sorry ;-)

File: source/DEC_API.c (hunk 2 above)
Comment (PhilippSchenk) [inline source/DEC_API.c]: @SebastianWolfarth No logging any more in DEC-API

**Comment by FloH11110** [comment]
resolve the conflicts and merge

**Comment by PhilippSchenk** [comment]
char replaced by uint16_ in new version...
merge now?

**Pull request merged**